### ISSUE TITLE
🐛 LSP Organize Imports action only available in some places in the document

### ISSUE THREAD

**Issue by rchl** [rome/tools#4097]
### Environment information

```block
CLI:
  Version:              11.0.0
  Color support:        true

Platform:
  CPU Architecture:     aarch64
  OS:                   macos
```

### What happened?

Whether the `Organize Imports (Rome)` code action is returned (in LSP server mode) depends on the location of the cursor in the document. For example in a document like:

```ts
/*
 * License
 */

import * as path from 'node:path';
import fs from 'fs-extra';
```

if I place the cursor within the comment block and trigger the `Source action...` from the context menu (in VSCode) then the `Organize Imports (Rome)` command is not provided. But if I do the same while the cursor is over the imports then it is provided. If there is more code below other than imports than the code action is also provided there.

### Expected result

The `Organize Imports (Rome)` code action should be provided anywhere within the document, irrespective of where "source action..." was triggered since it's a source-level code action (a `source.organizeImports.rome` kind) and shouldn't really depend on cursor location.

Is there any particular reason or statement in the LSP spec that justifies the current behavior?

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

**Comment by ematipico** [comment]
I don't think this is an issue related to Rome. While triaging the issue, I **disabled** Rome's extension, and the bug is triggered. 

Also, the bug occurs regardless of the comment block. This is probably an issue related to VSCode, or maybe it's the expected behaviour, I am not sure.

**Comment by rchl** [comment]
How could you reproduce it with Rome disabled when `Organize Imports (Rome)` is Rome-specific and doesn't show up with it disabled?

Also, I've reproduced this in ST initially.

**Comment by ematipico** [comment]
> How could you reproduce it with Rome disabled when Organize Imports (Rome) is Rome-specific and doesn't show up with it disabled?

Sorry, I did some mistakes while triaging. Are you aware of other tools that get this right?

**Comment by ematipico** [comment]
@leops you might want to take a look at this. Interestingly, the issue occurs only on top-level comments. Other comments still emit the correct context menu action.

**Comment by rchl** [comment]
> Are you aware of other tools that get this right?

You can reference default typescript language features in VSCode.
But generally "source" code actions should be available for the whole document and should not be specific to range that is included in the request (if even included).

**Comment by leops** [comment]
I think I have an idea of what the issue is, and I suspect it's highly specific to comments being at the very start of very end of the document. The `source.organizeImports` is being correctly emitted for the range of the whole document, but we're using the trimmed range of the document node that specifically does not include the comments and whitespace before the first token.
